main.go

```diff
@@ -82,6 +83,7 @@ func main() {
 	flag.BoolVar(&watchAllNamespaces, "watch-all-namespaces", true,
 		"Watch for custom resources in all namespaces, if set to false it will only watch the runtime namespace.")
 	flag.DurationVar(&rateLimitInterval, "rate-limit-interval", 5*time.Minute, "Interval in which rate limit has effect.")
+	flag.BoolVar(&insecureNoTLS, "insecure-no-tls", false, "Enable the use of HTTP Scheme (no TLS) across all controller level objects. This is not recommended for production environments")
```

Comment:
Let's not introduce this as breaking change now, we will revisit this in the future.
```suggestion
	flag.BoolVar(&insecureNoTLS, "insecure-no-tls", true, "Enable the use of HTTP Scheme (no TLS) across all controller level objects. This is not recommended for production environments")
```

Reply:
If I understand this PR correctly than this flag's name is hard to grasp and I'd recommend this instead to clarify that this pertains to HTTP vs. HTTPS:
```suggestion
	flag.BoolVar(&insecureAllowHTTP, "insecure-allow-http", false, "Enable the use of HTTP Scheme (no HTTPS) across all controller level objects. This is not recommended for production environments")
```
Alternative something like `insecureAllowPlainHTTP` would also work.